basic_bot.js

```diff
@@ -333,12 +334,35 @@ function script() {
 
   var getTarget = function (myX, myY, targetX, targetY, grid) {
         // TODO: handle edge cases regarding target and current position
-    var graph = new Graph(grid, {diagonal: true});
+    // var graph = new Graph(grid, {diagonal: true});
+    var graph = new Graph(grid);
     var start = graph.grid[myX][myY];
     var end = graph.grid[targetX][targetY];
-    var shortestPath = astar.search(graph, start, end, { heuristic: astar.heuristics.diagonal });
-        // var shortestPath = astar.search(graph, start, end);
-    var next = shortestPath[0];
+    // var shortestPath = astar.search(graph, start, end, { heuristic: astar.heuristics.diagonal });
+    var shortestPath = astar.search(graph, start, end);
+
+    // Find the furthest cell in the direction of the next cell
+    var winner = 0;
+    if (shortestPath.length > 1) {
+      var dx = shortestPath[0].x - myX;
+      var dy = shortestPath[0].y - myY;
+      for (var i = 0; i < shortestPath.length; i++) {
```

Comment:
Do you prefer `break`ing a for loop rather than having a while loop conditional on something called `straightawayContinues` or something? Either way is fine I think - just checking.